parquet-hadoop/src/test/java/org/apache/parquet/crypto/propertiesfactory/SchemaCryptoPropertiesFactory.java

```diff
@@ -0,0 +1,135 @@
+
+  @Override
+  public FileEncryptionProperties getFileEncryptionProperties(Configuration conf, Path tempFilePath,
+                                                              WriteContext fileWriteContext) throws ParquetCryptoRuntimeException {
+    MessageType schema = fileWriteContext.getSchema();
+    List<String[]> paths = schema.getPaths();
+    if (paths == null || paths.isEmpty()) {
+      throw new ParquetCryptoRuntimeException("Null or empty fields is found");
+    }
+
+    Map<ColumnPath, ColumnEncryptionProperties> columnPropertyMap = new HashMap<>();
+
+    for (String[] path : paths) {
+      getColumnEncryptionProperties(path, columnPropertyMap, conf);
+    }
+
+    if (columnPropertyMap.size() == 0) {
+      log.debug("No column is encrypted. Returning null so that Parquet can skip. Empty properties will cause Parquet exception");
+      return null;
+    }
+
+    /**
+     * Why we still need footerKeyMetadata even withEncryptedFooter as false? According to the
+     * 'Plaintext Footer' section of
+     * https://github.com/apache/parquet-format/blob/encryption/Encryption.md, the plaintext footer
+     * is signed in order to prevent tampering with the FileMetaData contents. So footerKeyMetadata
+     * is always needed. This signature will be verified if parquet-mr code is with parquet-1178.
+     * Otherwise, it will be ignored.
+     */
+    boolean shouldEncryptFooter = getEncryptFooter(conf);
+    FileEncryptionProperties.Builder encryptionPropertiesBuilder =
+      FileEncryptionProperties.builder(FOOTER_KEY)
+        .withFooterKeyMetadata(FOOTER_KEY_METADATA)
+        .withAlgorithm(getParquetCipherOrDefault(conf))
+        .withEncryptedColumns(columnPropertyMap);
+    if (!shouldEncryptFooter) {
+      encryptionPropertiesBuilder = encryptionPropertiesBuilder.withPlaintextFooter();
+    }
+    FileEncryptionProperties encryptionProperties = encryptionPropertiesBuilder.build();
+    log.info(
+      "FileEncryptionProperties is built with, algorithm:{}, footerEncrypted:{}",
+      encryptionProperties.getAlgorithm(),
+      encryptionProperties.encryptedFooter());
+    return encryptionProperties;
+  }
+
+  private ParquetCipher getParquetCipherOrDefault(Configuration conf) {
+    String algorithm = conf.get(CONF_ENCRYPTION_ALGORITHM, "AES_GCM_CTR_V1");
+    log.debug("Encryption algorithm is {}", algorithm);
+    return ParquetCipher.valueOf(algorithm.toUpperCase());
+  }
+
+  private boolean getEncryptFooter(Configuration conf) {
+    boolean encryptFooter = conf.getBoolean(CONF_ENCRYPTION_FOOTER, false);
+    log.debug("Encrypt Footer: {}", encryptFooter);
+    return encryptFooter;
+  }
+
+  private void getColumnEncryptionProperties(String[] path, Map<ColumnPath, ColumnEncryptionProperties> columnPropertyMap,
+                                             Configuration conf) throws ParquetCryptoRuntimeException {
+    String pathName = String.join(".", path);
+    String columnKeyName = conf.get(pathName, null);
```

Comment:
I don't think it is a good practice to use the column name as a conf key directly. The chance of collisions are pretty high. I would suggest adding a constant prefix.

Reply:
fixed